src/libstd/io/tempfile.rs

```diff
@@ -78,47 +80,59 @@ pub struct TempDir {
     disarmed: bool
 }
 
+// How many times should we (re)try finding an unused random name? It should be
+// enough that an attacker will run out of luck before we run out of patience.
+const NUM_RETRIES: uint = 1 << 31;
+// How many characters should we include in a random file name? It needs to
+// be enough to dissuade an attacker from trying to preemptively create names
+// of that length, but not so huge that we unnecessarily drain the random number
+// generator of entropy.
+const NUM_RAND_CHARS: uint = 12;
```

Comment:
`Iterator::take` takes a `uint`, and this is just passed straight to `take`. It may or may not be correct for `take` to take a `uint`, but considering the signature of `take` isn't part of this PR.

If `take` is changed as part of the examine-integers stdlib pass that we're planning, then it will make sense for this constant to not be a `uint` and it will be updated then.